service/auth.go

```diff
@@ -120,6 +129,14 @@ func (s *Service) unregisterClient(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// an authedClientID == "" means admin. So if there is an authedClientID,
+	// then the requested clientID needs to be the same.
+	if authedClientID != "" && authedClientID != clientID {
+		data.err = "client id not valid"
+		data.code = http.StatusBadRequest
```

Comment:
I think this case would fit better in the 403 code since there's nothing wrong with the request itself but the server simply cannot go ahead due to missing authorization. Not a big deal though.

Reply:
Yep, makes sense, thanks!